File: src/py/tests/error-handling.test.py
from styled_terminal import Style, Color, InvalidParameter
import unittest

class TestColorErrorHandling(unittest.TestCase):
    def test_invalid_rgb_values(self):
        with self.assertRaises(InvalidParameter):
            Color.rgb(300, 0, 0)
    
    def test_invalid_hex_code(self):
        with self.assertRaises(InvalidParameter):
            Color.hex("invalid")
    
    def test_invalid_256_color_index(self):
        with self.assertRaises(InvalidParameter):
            Color.table256(-1)

if __name__ == "__main__":
    unittest.main()

File: src/js/tests/error-handling.test.ts
import { Color } from "../src/index";
import { InvalidParameter } from "../src/errors";

describe("Error Handling", () => {
  it("should throw an error for invalid RGB values", () => {
    expect(() => Color.rgb(300, 0, 0)).toThrow(InvalidParameter);
  });

  it("should throw an error for invalid hex codes", () => {
    expect(() => Color.hex("invalid")).toThrow(InvalidParameter);
  });

  it("should throw an error for invalid 256-color index", () => {
    expect(() => Color.table256(-1)).toThrow(InvalidParameter);
  });
});
